### PR TITLE
Cover more route line placement edge cases

### PR DESCRIPTION
**Pull request by LukasPaczos**
Closes https://github.com/mapbox/mapbox-navigation-android/issues/3700.

The issue was that `SymbolLayer`s are not always on top of the layer stack. They can be buried deep down under other layers, especially labels for tunnels and underpasses. This means that we cannot place the route line below the first `SymbolLayer`, it has to be above the first non-symbol layer instead which the previous implementation tried to cover but I failed to notice it. This edge case is now covered with a test.

### CODE REVIEW

File: libnavigation-ui/src/main/java/com/mapbox/navigation/ui/route/MapRouteLine.kt [hunk 1]
```diff
@@ -1254,12 +1254,35 @@ internal class MapRouteLine(
...
          */
         fun getBelowLayer(layerId: String?, style: Style): String {
+            val layers = style.layers
             return when (layerId.isNullOrEmpty()) {
-                false -> style.layers.firstOrNull { it.id == layerId }?.id
-                true -> style.layers.firstOrNull {
-                    it.id.contains(RouteConstants.MAPBOX_LOCATION_ID) || it is SymbolLayer
-                }?.id
-            } ?: LocationComponentConstants.SHADOW_LAYER
+                // if a desired ID was provided, try to find it
```
Comment (Guardiola31337) [inline libnavigation-ui/src/main/java/com/mapbox/navigation/ui/route/MapRouteLine.kt:1259]: What about extracting this block of code into a `private` `fun`ction and give it a name based on the comment? This way `getBelowLayer` will be easier to read and understand and comment will become superfluous so it won't be necessary.

File: libnavigation-ui/src/main/java/com/mapbox/navigation/ui/route/MapRouteLine.kt [hunk 2]
```diff
@@ -1254,12 +1254,35 @@ internal class MapRouteLine(
...
+                    if (it == null) {
+                        Timber.e(
+                            """Tried placing route line below "$layerId" which doesn't exist"""
+                        )
+                    }
+                }
+                // If there's no externally provided ID,
+                // then place the route line below the location puck but only if
+                // the puck is not mixed with other symbol layers on top of the stack.
+                // Otherwise, place the line above the first non-symbol layer.
```
Comment (Guardiola31337) [inline libnavigation-ui/src/main/java/com/mapbox/navigation/ui/route/MapRouteLine.kt:1270]: Same as above re: comments

File: libnavigation-ui/src/test/java/com/mapbox/navigation/ui/route/MapRouteLineTest.kt [hunk 3]
```diff
@@ -642,6 +675,20 @@ class MapRouteLineTest {
         assertEquals(LocationComponentConstants.SHADOW_LAYER, result)
     }
 
+    @Test
+    fun getBelowLayerWithNullLayerId_noSymbolLayers_2() {
```
Comment (Guardiola31337) [inline libnavigation-ui/src/test/java/com/mapbox/navigation/ui/route/MapRouteLineTest.kt:679]: NIT What's the difference between `getBelowLayerWithNullLayerId_noSymbolLayers` and `getBelowLayerWithNullLayerId_noSymbolLayers_2`? Could we name it something different instead `2` to provide more context as to what's being tested?

**Review by Guardiola31337 — COMMENTED** [review]
Thanks for the quick fix @LukasPaczos 

Left some minor comments.

**Comment by LukasPaczos** [comment]
Ready or another round.

**Review by Guardiola31337 — APPROVED** [review]
Thanks for addressing the feedback 🚀
